src/sidebar/util/version-data.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+/**
+ * An object representing user info
+ *
+ * @typedef {Object} UserInfo
+ * @property {string=} userid
+ * @property {string=} displayName
+ */
+
+/**
+ * An object representing document metadata
+ *
+ * @typedef {Object} DocMetadata
+ * @property {string=} documentFingerprint - optional PDF fingerprint for
+ *                     current document
+ */
+
+/**
+ * An object representing document info
+ *
+ * @typedef {Object} DocumentInfo
+ * @property {string=} url - current document URL
+ * @property {DocMetadata} metadata - document metadata
+ */
+
+class VersionData {
+  /**
+   * @param {UserInfo} userInfo
```

Comment:
I think they're useful because they declare the `type` (`UserInfo`, e.g.) of the parameters, which isn't information that is obvious in any other way.